cmd/cloudinitsave/cloudinitsave.go

```diff
@@ -146,9 +116,67 @@ func saveCloudConfig() error {
 	return saveFiles(userDataBytes, scriptBytes, metadata)
 }
 
-func fetchUserData() ([]byte, datasource.Metadata, error) {
+func RequiresNetwork(datasource string) bool {
+	parts := strings.SplitN(datasource, ":", 2)
+	requiresNetwork, ok := map[string]bool{
+		"ec2":          true,
+		"file":         false,
+		"url":          true,
+		"cmdline":      true,
+		"configdrive":  false,
+		"digitalocean": true,
+		"gce":          true,
+		"packet":       true,
+	}[parts[0]]
+	return ok && requiresNetwork
```

Comment:
I don't see this being an issue. If there's a specific use case brought up by users later we can definitely address it then.